corgidrp/walker.py

```diff
@@ -95,12 +103,23 @@ def autogen_recipe(filelist, outputdir, template=None):
         if "calibs" in step:
             for calib in step["calibs"]:
                 # order matters, so only one calibration file per dictionary
-
                 if step["calibs"][calib].upper() == "AUTOMATIC":
                     calib_dtype = data.datatypes[calib]
                     best_cal_file = this_caldb.get_calib(first_frame, calib_dtype)
                     # set calibration file to this one
                     step["calibs"][calib] = best_cal_file.filepath
+
+                # changed to handle the master dark and master flat files, but this assumes there is only one of each type 
```

Comment:
I modified the code to eliminate the need to pass a dummy frame into the walker. Since the bp map depends solely on noise maps and flat fields, introducing a dummy frame seemed unnecessary. The function section handling the "AUTOMATIC" keyword uses get_calib, which requires an input frame from the same timeframe as the master dark and flat. Should we assume that the AUTOMATIC selection would work here, or might there be instances where the dark and flat were produced at different times, complicating this assumption?